oauthenticator/oauth2.py

```diff
@@ -663,6 +680,17 @@ def _allowed_scopes_validation(self, proposal):
         """,
     )
 
+    require_pkce = Bool(
+        False,
+        config=True,
+        help="""
+        Require Proof Key for Code Exchange (PKCE) for OAuth2 authorization code flow.
```

Comment:
Sorry for giving you the runaround a bit on names, but let's use:

```suggestion
    enable_pkce = Bool(
        True,
        config=True,
        help="""
        Enable Proof Key for Code Exchange (PKCE) for OAuth2 authorization code flow.
```

I think `require_pkce` gives the wrong impression about what oauthenticator is in a position to enforce.